daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetImpl.scala

```diff
@@ -443,7 +443,7 @@ sealed abstract class LengthState(ie: DIElement)
 
   def isEndUndef = {
     val r = maybeEndPos0bInBits.isEmpty
-    if (r) Assert.invariant(maybeStartDataOutputStream.isEmpty)
+    if (r) Assert.invariant(maybeEndDataOutputStream.isEmpty)
```

Comment:
Is this just a bug fix? Was this being called?

Reply:
I think this is just a bug that got triggered by evaluating suspensions earlier. If the end position is empty, thent he end output stream must also be empty, not the start output stream. I suspect this was just a copy/paste error from isStartUndef.

Reply:
To your second question, I think it was being called, it just always happend that the start output strema was empty if the end pos was empty due to how suspensions were evaluated.